src/pretix/presale/views/event.py

```diff
@@ -104,6 +108,29 @@ def get_context_data(self, **kwargs):
         return context
 
 
+class EventIcalDownload(EventViewMixin, View):
+    def get(self, request, *args, **kwargs):
+        if not self.request.event:
+            raise Http404(_('Unknown event code or not authorized to access this event.'))
+
+        cal = Calendar()
+        cal.add('version', '2.0')
+        cal.add('prodid', '-//{}//{}//'.format(self.request.event.organizer.name, self.request.event.name))
+
+        event = Event()
+        event.add('summary', self.request.event.name)
+        event.add('dtstart', self.request.event.date_from)
```

Comment:
According to icalendar document, I only need to provide datetime object for icalendar to generate the correct timestamp.[http://icalendar.readthedocs.io/en/latest/usage.html#components](doc)

However, you also mentioned timezone. My bad :( The original version is wrong!
Now I fixed it and it will output
```
DTSTART;TZID=Asia/Taipei;VALUE=DATE-TIME:20170220T073456
```
which is the same as the icalendar spec with timezone specified if it isn't UTC. (It's handled by `icalendar` automatically)
[https://icalendar.org/iCalendar-RFC-5545/3-6-1-event-component.html](DTSTART)
[https://icalendar.org/iCalendar-RFC-5545/3-2-19-time-zone-identifier.html](TZID)

I would like to commit my changes after confirming the behaviours of the event settings.
For `self.request.event.settings.show_times`, it's easy since `icalendar` supports date object too!

For `self.request.event.settings.show_date_to`, it's a little bit complicated.
I went through the specs of icalendar.
[https://icalendar.org/iCalendar-RFC-5545/3-6-1-event-component.html](DTSTART/DTEND)

**Case 1)**
If we provide datetime `DTSTART` without `DTEND`, end date will be the same as the start date. 
> For cases where a "VEVENT" calendar component specifies a "DTSTART" property with a DATE-TIME value type but no "DTEND" property, the event ends on the same calendar date and time of day specified by the "DTSTART" property.

So if `self.request.event.settings.show_times is True and self.request.event.settings.show_date_to is False`,
I guess we can just provide DTSTART so that the time can be kept.

**Case 2)**
If we provide date `DTSTART` without `DTEND`, the event will become an one-day event.
> For cases where a "VEVENT" calendar component specifies a "DTSTART" property with a DATE value type but no "DTEND" nor "DURATION" property, the event's duration is taken to be one day.

If `self.request.event.settings.show_times is False and self.request.event.settings.show_date_to is False`,
we will set `DTSTART` date and the event lasts for one day.

**Case 3)**
If `self.request.event.settings.show_times is False and self.request.event.settings.show_date_to is True`, we will ignore timezone and set `date` in both `DTSTART` and `DTEND`.

Does it sound good? Thanks for your comments :)